server/graphql/catalogue/src/asset_catalogue_item_queries.rs

```diff
@@ -0,0 +1,173 @@
+use async_graphql::*;
+use graphql_core::{
+    generic_filters::{EqualFilterStringInput, StringFilterInput},
+    pagination::PaginationInput,
+    simple_generic_errors::{NodeError, NodeErrorInterface},
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+use graphql_types::types::{
+    AssetCatalogueItemConnector, AssetCatalogueItemNode, AssetCatalogueItemResponse,
+    AssetCatalogueItemsResponse,
+};
+use repository::asset_catalogue_item::{
+    AssetCatalogueItemFilter, AssetCatalogueItemSort, AssetCatalogueItemSortField,
+};
+use repository::{EqualFilter, PaginationOption, StringFilter};
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    catalogue::query_catalogue_item::{get_asset_catalogue_item, get_asset_catalogue_items},
+};
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq)]
+#[graphql(remote = "repository::asset_catalogue_item::AssetCatalogueItemSortField")]
+#[graphql(rename_items = "camelCase")]
+
+pub enum AssetCatalogueItemSortFieldInput {
+    Catalogue,
+    Code,
+    Make,
+    Model,
+}
+
+#[derive(InputObject)]
+
+pub struct AssetCatalogueItemSortInput {
+    key: AssetCatalogueItemSortFieldInput,
+    desc: Option<bool>,
+}
+
+#[derive(InputObject, Clone)]
+
+pub struct AssetCatalogueItemFilterInput {
+    pub id: Option<EqualFilterStringInput>,
+    pub category: Option<StringFilterInput>,
+    pub category_id: Option<EqualFilterStringInput>,
+    pub class: Option<StringFilterInput>,
+    pub class_id: Option<EqualFilterStringInput>,
+    pub code: Option<StringFilterInput>,
+    pub manufacturer: Option<StringFilterInput>,
+    pub model: Option<StringFilterInput>,
+    pub r#type: Option<EqualFilterStringInput>,
```

Comment:
```suggestion
    pub r#type: Option<StringFilterInput>,
```
I think the idea here is to be able search for a partial type name, eg. `Freezer` which would find `Fridge/Freezer` as well as `Freezer` etc...

Reply:
`type_id` is the exact match one